### PR TITLE
add test for const-eval error in dead code during monomorphization

### PR DESCRIPTION
**Pull request by RalfJung**

### CODE REVIEW

File: tests/ui/consts/const-eval/unused-broken-const-late.stderr [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+LL |     const VOID: () = panic!();
+   |                      ^^^^^^^^ the evaluated program panicked at 'explicit panic', $DIR/unused-broken-const-late.rs:7:22
+   |
+   = note: this error originates in the macro `$crate::panic::panic_2015` which comes from the expansion of the macro `panic` (in Nightly builds, run with -Z macro-backtrace for more info)
+
+note: the above error was encountered while instantiating `fn no_codegen::<i32>`
+  --> $DIR/unused-broken-const-late.rs:18:5
+   |
+LL |     no_codegen::<i32>();
+   |     ^^^^^^^^^^^^^^^^^^^
```
Comment (RalfJung) [inline tests/ui/consts/const-eval/unused-broken-const-late.stderr:13]: @oli-obk ideally the error would point not only here, but also at line 14 where the constant syntactically appears. Any idea why it doesn't do that?

Comment (oli-obk) [reply tests/ui/consts/const-eval/unused-broken-const-late.stderr:13]: hmm... good point ^^ probably an oversight in the mono time error handler

**Comment by RalfJung** [comment]
In fact whether the span at `no_codegen::<i32>();` is shown seems to depend on the optimization level?

**Pull request merged**